### DISCUSSION TITLE
Test error on Android 6.0

### DISCUSSION THREAD

**Discussion by design-study** [trojan-gfw/igniter#381] [Suspicious Problems]
When I run on the Android 6 emulator, I can connect to the server, but the test will fail.

```
04-13 09:43:57.639 4367-4367/io.github.trojan_gfw.igniter.debug E/MainActivity: TestError: recvfrom failed: ECONNRESET (Connection reset by peer)
```

Environment:

* Android SDK version: 23
* Emulator CPU/ABI: x86_64
* igniter version: [c2261a97](https://github.com/trojan-gfw/igniter/commit/c2261a97406f0fa84987e3cbfbed6dfcfaff3ca5)
* igniter-go-libs version: [72ba215](https://github.com/trojan-gfw/igniter-go-libs/commit/72ba215843cf9c25ef13a14b641dd7410fe1bf36)
* libs version: [8fde414](https://github.com/trojan-gfw/igniter-libs/tree/8fde414e3ebf0f28ab40f5835ad0d47ee85758c5)
* trojan version: [8606b71](https://github.com/trojan-gfw/trojan/tree/8606b7110fe79f8ab02d60c897f87ffb0a9b23f0)

**Comment by wongsyrone** [comment]
Please obtain the log via instructions in https://github.com/trojan-gfw/igniter/issues/197

**Comment by design-study** [comment]
@wongsyrone 

After I click the `START` button, I can access the website through the proxy normally, but when I click the `Test Connection` button, it will display `recvfrom failed: ECONNRESET (Connection reset by peer)`, and this situation will only appear in Android 6.0 (API 23) on the device.

Log detail:

[igniter-log.txt](https://github.com/trojan-gfw/igniter/files/6355160/igniter-log.txt)

**Comment by wongsyrone** [comment]
@design-study Please post your kernel version

**Comment by design-study** [comment]
@wongsyrone 

![Screenshot_1619172230](https://user-images.githubusercontent.com/7581969/115861100-f6d87d80-a464-11eb-839a-68f047032e3b.png)

**Comment by wongsyrone** [comment]
> Depending on the environment, the emulator may not be able to support other protocols (such as ICMP, used for "ping") might not be supported. 

https://www.androidcookbook.info/android-system/emulator-networking.html

I cannot reproduce the issue.

**Comment by design-study** [comment]
The Android 6.0 (API 23) emulator should be fine. After the service is connected, you can access the Google search homepage opening the browser. Using the adb shell to detect, the simulator can execute the ping command.

The problem is that clicking the test button in the app reports an error.

**Comment by hoorace** [comment]
I encountered the same problem, this is my device information.


![Screenshot](https://user-images.githubusercontent.com/345589/118439227-1f782e00-b718-11eb-9566-0ffa54360374.png)
